Version 2.4: UI improvement

Sevoos opened the pull request:
Fixed a minor spelling bug ("Удалить ведушего" -> "Удалить ведущего") 
Implemented "reveal roles" mode for accounts: prevents unintentional roles reveal when the roles are shuffled and shown

Sevoos committed:
Version 2.4: UI improvement

Fixed a minor spelling bug ("Удалить ведушего" -> "Удалить ведущего")
Implemented "reveal roles" mode for accounts: prevents unintentional roles reveal when the roles are shuffled and shown

VladoCC commented on src/main/kotlin/Data.kt:
```diff
@@ -14,7 +14,8 @@ data class Account(
     var menuMessageId: Long = -1L,
-    var connectionId: ConnectionId? = null
+    var connectionId: ConnectionId? = null,
+    var revealRolesMode: Boolean = true
```
This is definitely not a place for storing this
Account is an object describing general user, who's not even guaranteed to ever be a host

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -709,6 +709,13 @@ fun showRoles(
         }
         button(resetRolesCommand, game.id, messageId)
+        row {
+            button(blankCommand named "👁 Показ ролей")
```
Unclear

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -709,6 +709,13 @@ fun showRoles(
         button(resetRolesCommand, game.id, messageId)
+        row {
+            button(blankCommand named "👁 Показ ролей")
+            button(
```
Checkbox looks out of place in a menu that does not use checkboxes

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -741,7 +748,13 @@ fun showPreview(
+                    } else {
+                        "✅ Роль выдана"
+                    }
+                } ?: "Роль не выдана"))
```
If one option of this button has an emoji then the other one should to for coherency

VladoCC replied:
I would prefer 👌 and ❗️
Ok is less intrusive and exclamation marks something that needs attention, but not an error

VladoCC commented on src/main/kotlin/db/Db.kt:
```diff
@@ -61,11 +62,20 @@ class Collection<K: Any, T: Any>(private val name: String, private val driver: D
+            e.printStackTrace()
+            null
+        }
+        // У уже созданных аккаунтов по умолчанию должно быть true, а gson делает false
```
no comments in russian in code

VladoCC commented on src/main/kotlin/db/Db.kt:
```diff
@@ -61,11 +62,20 @@ class Collection<K: Any, T: Any>(private val name: String, private val driver: D
+            null
+        }
+        // У уже созданных аккаунтов по умолчанию должно быть true, а gson делает false
+        if (parsedObject is Account) {
```
This is definitely out of place, which this check for a single type shows
Package `db` contains code of a general-purpose database. No project specific code is allowed here, especially object specific.

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -709,6 +709,13 @@ fun showRoles(
+            button(blankCommand named "👁 Показ ролей")
+            button(
+                toggleRevealRolesModeCommand named if (accounts.get(chatId)!!.revealRolesMode) "✅" else "❌",
+                chatId, messageId
```
First parameter is never used

Sevoos committed:
Version 2.4.1: logical rework of the roles revelation

Roles can now be revealed by specifically activating a new button in the "showPreview" menu
The status of role assignment now has emojis "okay" and "exclamation"
No data classes have a "revealRolesMode" field anymore (fix 2.4)
The "showPreview" function now has an optional Boolean parameter "revealRolesMode", false by default
Object-specific junk has been removed from Db.kt (fix 2.4)

Sevoos committed:
Version 2.4.2: rollback one accidental extra tab

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -725,7 +725,8 @@ fun showPreview(
     messageId: Long,
-    game: Game
+    game: Game,
+    revealRolesMode: Boolean = false
```
Breaks original behavior by reversing it

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -750,6 +757,13 @@ fun showPreview(
         row {
             button(blankCommand named "Распределено ролей: ${pairs.size}")
         }
+        button(
```
Would not work, because opening from any other menu would result in showing roles
Or it would be infuriating when you don't need to hide roles, since you'd have to click this button every time 
Either way, one side or another would dislike this approach

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -1065,6 +1065,12 @@ object MafiaHandler {
                 }
                 showRoles(chatId, long(1), bot, game)
             }
+            parametrized(enableRevealRolesModeCommand) {
```
Still could've been done with one button

VladoCC commented:
Reveal menu is not covered in any version, which is a major issue for a proposed use case.

Sevoos committed:
Version 2.4.3: Logical rework of roles revelation

The "revealRolesMode" field now belongs to data class Game
The switch button to toggle the mode is now available from both lobby and preview menus
The implementation works with one single command

VladoCC commented on src/main/kotlin/Data.kt:
```diff
@@ -35,7 +35,8 @@ data class Game(
     val creatorId: Long = hostId,
-    var actual: Boolean = true
+    var actual: Boolean = true,
+    var revealRolesMode: Boolean = true
```
This still means that every time host starts a game, they have to push this button if they need to hide roles. There are existing ways in a project to keep such settings persistent, so that host have to push this button only when they want to change state

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -929,6 +928,15 @@ object MafiaHandler {
             }
+            parametrized(toggleRevealRolesModeCommand) {
+                game.revealRolesMode = !game.revealRolesMode
+                games.save(game)
```
Incorrect way of approaching an update. There are functions specifically intended for optimised updates of   database objects

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -929,6 +928,15 @@ object MafiaHandler {
+                game.revealRolesMode = !game.revealRolesMode
+                games.save(game)
+                if (game.state == GameState.Connect) {
+                    showLobbyMenu(bot = bot, chatId = chatId, messageId = messageId!!, game = game)
```
Don't understand, why is it here?

Sevoos replied:
The point is that one command is invoked from 2 buttons in different menus (showPreview() and showLobbyMenu() functions in Main.kt). It means that it's needed to determine which of them to re-open, which can be done by checking for the GameState field. GameState.Connect corresponds to the lobby menu, GameState.Preview corresponds to the roles preview menu.

VladoCC replied:
I don't see two buttons in code

Sevoos replied:
[src/main/kotlin/Main.kt](https://github.com/VladoCC/AlmaMafia/pull/1/files#diff-2e9c104fc5144b79e630c4b1f599551d1b2f72462926691c2973b8e1d9da0b35) lines 759-764 for the preview menu and line 844 for the lobby menu

VladoCC replied:
This does not much sources I'm seeing in this PR
Github does not show any changes at 844 or around 800th line

Sevoos replied:
Not present anymore since 2.4.4

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -827,6 +839,10 @@ fun lobby(messageId: Long, game: Game): InlineKeyboardMarkup {
             button(changeHostCommand, messageId)
         }
+        row {
+            button(blankCommand named "👁 Предпросмотр ролей при раздаче")
```
Already wrote about checkboxes being out of place in this menu
Also, using `blankCommand` for a checkbox name is not the best UX practice.
Last thing, would this text even fit onto a button? Especially since it will be half length, that's part of the reason checkboxes aren't great for this.

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -929,6 +928,15 @@ object MafiaHandler {
+                if (game.state == GameState.Connect) {
+                    showLobbyMenu(bot = bot, chatId = chatId, messageId = messageId!!, game = game)
+                } else if (game.state == GameState.Preview) {
+                    showPreview(bot = bot, chatId = chatId, messageId = messageId!!, game = game)
```
`bot = bot, chatId = chatId, messageId = messageId!!, game = game` lowers readability because of doubled names

Sevoos committed:
Version 2.4.4: Logical rework of roles revelation

The "revealRolesMode" field now belongs to data class HostInfo
The switch button to toggle the mode is now available only from the preview menu
Fixed the collection element modification approach (manual changing with the save() function replaced with update() lambda expression)

VladoCC commented on src/main/kotlin/Data.kt:
```diff
@@ -259,7 +259,8 @@ data class HostInfo(
     var left: Int = -1,
-    var canShare: Boolean = true
+    var canShare: Boolean = true,
+    var revealRolesMode: Boolean = true
```
This class is not a place for storing host preferences. `HostInfo` is an access control class, which is clear by looking at other members of it.

Sevoos committed:
Version 2.4.5: Logical rework of roles revelation

The "revealRolesMode" field moved from hostInfos to HostSettings
Added the revealRolesAtPreview (corresponds to revealRolesMode) to the host settings menu

VladoCC commented on src/main/kotlin/Enums.kt:
```diff
@@ -75,6 +75,14 @@ enum class HostOptions(val shortName: String, val fullName: String, val current:
         }
+    ),
+    revealRolesAtPreview(
+        "👁 Предпросмотр ролей при раздаче",
```
This would result in showing your option in settings menu, which is not relevant to your feature and would just cause confusion

Sevoos replied:
It was intentional, but apparently a bad idea I guess

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -730,6 +730,7 @@ fun showPreview(
     val players = connections.find { gameId == game.id }
     val pairs = pairings.find { gameId == game.id }.associateBy { it.connectionId }
     val keyboard = inlineKeyboard {
+        val revealRolesMode = hostSettings.get(chatId)!!.revealRolesMode
```
Why is it so many blocks away from its usage? Especially since there is only one usage as far as i can see

Sevoos replied:
[src/main/kotlin/Main.kt](https://github.com/VladoCC/AlmaMafia/pull/1/files#diff-2e9c104fc5144b79e630c4b1f599551d1b2f72462926691c2973b8e1d9da0b35) field revealRolesMode at line 733 is used at both line 746 (inside a lambda expression, players.sortedby.forEach) and at line 762 (outside the lambda expression), that is the reason of its placement

VladoCC reviewed (changes requested):
Revel menu problem I mention a few iterations ago is still unsolved
Also, stop numbering versions randomly, there wasn't even one minor version yet between current code and this branch

Sevoos commented:
> Revel menu problem I mention a few iterations ago is still unsolved Also, stop numbering versions randomly, there wasn't even one minor version yet between current code and this branch

What is the current problem that needs to be fixed?

Sevoos committed:
2025-07-12 #4

Removed redundant switcher from Enums.kt

VladoCC commented:
Check for yourself by running it, there is a whole menu that is not covered by this change, which shows player roles

Sevoos committed:
2025-07-12 #5

"revealRolesMode" changed to "hideRolesMode", because gson makes a boolean false be default, therefore by default the roles are revealed, not hidden
Added mode switcher to the day menu
The mode switcher now impacts on the players list in the day menu

Sevoos committed:
2025-07-13 #1

Added a hide roles switch to the night menu
A separate command is used because it doesn't seem possible to determine whether it's day or night from Handling.kt parametrized function

Sevoos committed:
2025-07-13 #2

quick fix

Sevoos committed:
Revert "2025-07-13 #1"

This reverts commit 1e9bdb7aae9796f1ab622545b784cf7df003bc48.

Sevoos committed:
Revert "2025-07-13 #2"

This reverts commit 58e357a0bb60a441b4b628aff151ea5119bf4a37.

Sevoos committed:
2025-07-13 #3

quick fix

Sevoos committed:
2025-07-13 #4

Added the hideRolesMode switcher to the showRevealMenu() menu

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -179,15 +180,22 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
-                    button(if (leftName != null) blankCommand named leftName else blankCommand)
-                    val rightName = if (list.size < 2) null else list[1].role?.displayName
-                    button(if (rightName != null) blankCommand named rightName else blankCommand)
+                if (!hideRolesMode) {
```
This makes this menu almost useless since you don't know WHO didn't check their role

Sevoos replied:
If you mean that we don't know the *person* who didn't check their role, that's not the case. The hideRolesMode check there impacts only on the buttons that display roles. Did you mean that we don't know whether the murderer or a regular citizen (who doesn't have any abilities/actions) hasn't checked their role?

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -179,15 +180,22 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
             }
 
             button(blankCommand named "Ознакомлены: $notified / ${cons.size}")
+            button(
```
I would say that showing this button in reveal menu is quite redundant
I would prefer to keep this menu clean
If you want to roles to be hidden in the previous menu, I don't think you would change your mind in this one

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -425,6 +435,11 @@ internal fun showDayMenu(
                     }
                 }
             }
+            button(
```
I know that I've previously insisted on removing this button from settings menu, but if you are keeping it somewhere in day menu it should definitely be in settings menu instead
You should've explained why you put it there instead of moving it
Thinking about it, i agree that this button would be useful during the game if you died and now ready to fully host
Maybe even hide a button for starting night when roles are hidden, since you can't use night menus without seeing roles anyway

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -158,6 +158,7 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
             button(blankCommand named "Статус игроков")
             val cons = pairings.find { gameId == game.id }.sortedBy { it.connection?.pos ?: -1 }
             val notified = cons.count { it.connection?.notified ?: false }
+            val hideRolesMode = hostSettings.get(chatId)!!.hideRolesMode
```
Are we sure that `!!` would work at this point and existence is already a proven fact? Can't easily check it through diff view. Anyway, I would prefer `?:` 9 times out of 10. Especially since existence of this settings object wasn't proven in current scope (function) and relying on something outside of the scope is error-prone.

Also, straight up querying for host settings is a deprecated way of solving such problems, even though it is used everywhere in the code. Reason for it is just that most of the code is legacy and was written before access-methods were added. Instead of  `hostSettings.get(chatId)`, preferred way currently would be to use `game.host.settings` if I wrote it correctly. I think you get what I meant.

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -929,6 +928,21 @@ object MafiaHandler {
+                }
+                if (messageId != null) {
+                    val state = game.state
+                    if (state == GameState.Preview) {
```
First of all, this if-elseif tree screams `when (state)` to me.
Second of all, such a switch doesn't feel good to me anyway. I would prefer 3 different commands, correctly grouped with handlers of other buttons from the same menu and doing one specific thing. so, settings update goes to a function and update depends on command instead of if-elseif tree

VladoCC replied:
And I don't think we need this button twice in pregame menus, it only affects preview menu, so I would only expect to see it there

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -929,6 +928,21 @@ object MafiaHandler {
+                    if (state == GameState.Preview) {
+                        showPreview(bot, chatId, messageId, game)
+                    } else if (state == GameState.Game) {
+                        showDayMenu(towns[game.id]!!, chatId, messageId, bot, game)
```
Again, don't see a reason to use `!!` instead of a null-safe `let`

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -929,6 +928,21 @@ object MafiaHandler {
                 showLobbyMenu(chatId, long(0), game, bot)
             }
+            parametrized(toggleHideRolesModeCommand) {
+                hostSettings.update(chatId) {
```
There is a better way of doing that using `HostOptions`

VladoCC commented on src/main/kotlin/game/Game.kt:
```diff
@@ -308,11 +308,12 @@ internal fun joinGame(
             (if (!icons) "" else if (player.protected) "⛑️" else if (player.alive) "" else "☠️") +
             (if (!icons) "" else if (player.fallCount > 0) numbers[player.fallCount % numbers.size] else "") +
-            " ${player.name} (${player.roleData.displayName})"
+            " ${player.name} " +
```
Space after player name shouldn't be here, it should be in front of player role string

VladoCC commented on src/main/kotlin/game/Game.kt:
```diff
@@ -308,11 +308,12 @@ internal fun joinGame(
 }
 
-internal fun desc(player: Person?, sep: String = ". ", icons: Boolean = true) = if (player != null)
+internal fun desc(player: Person?, sep: String = ". ", icons: Boolean = true, hideRolesMode: Boolean = false) = if (player != null)
```
It feels like there would be places where this thing is used, which would reveal players role even though it supposed to be hidden
Not sure how to make it better yet, but I don't like just adding params to this function

VladoCC replied:
I still disagree with this one, would be better to split this functionality into multiple separate functions, but I'll let it pass

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -750,7 +757,12 @@ fun showPreview(
-        button(previewCommand named "🔄 Перераздать", game.id, messageId)
+        button(
+            toggleHideRolesModeCommand named
+                    if (hideRolesMode) "🐵 Показывать роли" else "🙈 Скрывать роли",
```
Check `hidePlayersCommand` for less intrusive emojis

VladoCC commented on src/main/kotlin/Commands.kt:
```diff
@@ -34,6 +34,7 @@ val acceptHostingCommand = command("Да", "acceptHosting", 3)
 val declineHostingCommand = command("Нет", "declineHosting", 3)
 
 val menuLobbyCommand = command("◀️ Меню игроков", "menuLobby", 1)
+val toggleHideRolesModeCommand = command("🙈 Скрывать роли", "toggleHideRolesMode", 1)
```
All commands are grouped by domain of usage, this one is out of place in between menu buttons

VladoCC commented on src/main/kotlin/Data.kt:
```diff
@@ -190,6 +190,7 @@ data class HostSettings(
     var timer: Boolean = false,
     var hideDayPlayers: Boolean = false,
     var playersHidden: Boolean = false,
+    var hideRolesMode: Boolean = true
```
Why is it true by default? I strongly disagree

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -179,15 +180,22 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
             button(blankCommand named "Ознакомлены: $notified / ${cons.size}")
+            button(
+                toggleHideRolesModeCommand named
+                        if (hideRolesMode) "🐵 Показывать роли" else "🙈 Скрывать роли",
```
You used the same if-else for names in 3 different places. Extract it.

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -404,14 +412,16 @@ internal fun showDayMenu(
                                 (if (settings?.detailedView == true) blankCommand else dayDetailsCommand) named desc(
-                                    player
+                                    player,
+                                    hideRolesMode = hideRolesMode
```
That's exactly why I disagree with adding it as a function parameter

Sevoos replied:
I personally don't see a better way of expressing that. It is probably not *too* unreadable for a single double-writing in one method invoking. Meanwhile, the desc() function doesn't have any other parameters that could determine the hideRolesMode

Sevoos committed:
2025-07-17 #1

Logical rework of roles revelation

VladoCC commented on src/main/kotlin/Commands.kt:
```diff
@@ -53,6 +53,7 @@ val decrCommand = command("➖", "decr", 2)
 val resetRolesCommand = command("🚮 Сбросить выбор ролей", "resetRoles", 2)
 val previewCommand = command("🔀 Раздать роли", "preview", 2)
 val gameModeCommand = command("", "mode", 2)
+val toggleHideRolesModePreviewCommand = command("😑 Скрывать роли", "toggleHideRolesModePreview", 1)
```
Still not quite what I would like to see here
Difference in emoji is too hard to see and it is still feels intrusive

Sevoos replied:
Would it be a good idea to leave it pure text, without emojis?

VladoCC replied:
No, I've pointed out that `hidePlayersCommand` has perfect emoji for this

VladoCC commented on src/main/kotlin/Commands.kt:
```diff
@@ -65,6 +66,7 @@ val dayBackCommand = command("◀️ Назад", "dayBack", 1)
 val settingsCommand = command(/*📋*/"⚙️ Настройки", "settings", 1)
 val timerCommand = command("⏳ Таймер", "timer")
 val nightCommand = command("🌙 Начать ночь", "night", 1)
+val startRevealingRolesCommand = command(name = "😐 Начать показывать роли", "startRevealingRoles", 1)
```
What's going on with button texts? Why is it different and so long for this one and not the same as the other one that does the same general thing.

VladoCC commented on src/main/kotlin/Data.kt:
```diff
@@ -35,7 +35,8 @@ data class Game(
     val creatorId: Long = hostId,
-    var actual: Boolean = true
+    var actual: Boolean = true,
+    var overrideHideRolesMode: Boolean = false
```
Why would we override something? You are returning to your previous solution, which I didn't agree with. HostSetting option does more than enough.

Sevoos replied:
This is a variable that is set to true once a formerly alive player becomes the host. It makes it possible not to hide roles without changing the hostSettings configuration. Isn't that a good approach?

VladoCC replied:
No, it does not seem good to me. It's a second boolean that does the same thing as the original one, producing ambiguity in place where just changing setting is enough.

Sevoos replied:
As the shouldHideRoles method suggests, once the overrideHideRolesMode variable is set to true (it is unavoidably true after day 1), the roles are never hidden. However, while it stays false, roles are either hidden or revealed, depending on the hostSettings configuration. I designed it like that because otherwise (if we just modify hostSettings during the game), hideRolesMode stays true at the end of the game, even though it was false before the game had began.

VladoCC replied:
I don't see a problem in it staying false, but I see problem in forced true for the whole game, which I pointed in another comment

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -328,6 +331,14 @@ fun showPlayerDayDesc(town: Town, playerPos: Int, messageId: Long, chatId: Long,
+    return ((game.host ?: accounts.get(chatId)!!).settings ?: hostSettings.get(chatId)!!).hideRolesMode
+}
+
+fun shouldHideRoles(game: Game, chatId: Long): Boolean {
```
This is super messy

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -328,6 +331,14 @@ fun showPlayerDayDesc(town: Town, playerPos: Int, messageId: Long, chatId: Long,
 }
 
+fun getHideRolesMode(game: Game, chatId: Long): Boolean {
+    return ((game.host ?: accounts.get(chatId)!!).settings ?: hostSettings.get(chatId)!!).hideRolesMode
```
What is going on here? That's not what I said to use for accessing db data. It tries to find both host and settings objects twice instead of doing it just once.

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -317,7 +317,10 @@ fun showPlayerDayDesc(town: Town, playerPos: Int, messageId: Long, chatId: Long,
-                button(dayDetailsCommand named desc(player), playerPos, messageId)
+                button(dayDetailsCommand named desc(
+                    player,
+                    hideRolesMode = shouldHideRoles(games.get(town.gameId)!!, chatId)
```
`!!` again.

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -1178,6 +1183,18 @@ object MafiaHandler {
             }
+            parametrized(startRevealingRolesCommand) {
+                games.update(game.id) {
+                    overrideHideRolesMode = true
```
You are always just one misclick away from revealing all the roles without a single chance to hide them back

VladoCC commented on src/main/kotlin/telegram/Menu.kt:
```diff
@@ -429,7 +433,11 @@ internal fun showDayMenu(
             }
-            button(nightCommand, msgId)
+            if (hideRolesMode) {
+                button(startRevealingRolesCommand, msgId)
```
I've said that I would like to see this thing back in settings menu and not cluttering main game menu. Less clutter = more players fitting on smaller screen.

Sevoos committed:
2025-07-18_01

VladoCC commented on src/main/kotlin/Const.kt:
```diff
@@ -2,7 +2,7 @@ package org.example
 object Const {
     object Message {
-        const val numSaved = "Номер игрока сохранен"
+        const val numSaved = "Номер игрока сохранён"
```
Don't like this pedantic approach to `e` characters

VladoCC commented on src/main/kotlin/Enums.kt:
```diff
@@ -75,6 +75,14 @@ enum class HostOptions(val shortName: String, val fullName: String, val current:
         }
+    ),
+    HideRolesMode(
+        "🕶 Скрывать роли",
```
Sad that this names are not adaptive, would be a cool improvement to replace string with lambda that receives setting value and returns name
Out of scope for this PR tho

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -328,6 +332,16 @@ fun showPlayerDayDesc(town: Town, playerPos: Int, messageId: Long, chatId: Long,
+    } ?: false
+}
+
+fun getHideRolesMode(gameId: GameId): Boolean {
```
What is this? if you have game everywhere, just use the first function, if you don't then use second one
But I don't see muсh reason for any of it
Especially of the only reason is that you have town instead of a game at some places. Just pace a game and retrieve town from it instead.

Sevoos replied:
You probably meant "pace town and retrieve game", but your request is totally valid, since game: Game? is nullable

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -750,7 +771,12 @@ fun showPreview(
-        button(previewCommand named "🔄 Перераздать", game.id, messageId)
+        button(
+            toggleHideRolesModePreviewCommand named
+                    if (hideRolesMode) "👓 Показывать роли" else toggleHideRolesModePreviewCommand.name,
```
Would be more readable in my opinion if both sides of this if are plain strings. Copying one string isn't that big of a deal.

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -1179,23 +1182,40 @@ object MafiaHandler {
+                        val res = bot.sendMessage(
                             ChatId.fromId(chatId),
-                            "Результат дня:\n${shortLog(town).ifBlank { "Не произошло никаких событий" }}"
+                            "Выключите скрытие ролей в меню настроек"
```
Super unnecessary. Just hide `nightCommand` button until they reveal roles and that's it

Sevoos replied:
It may be unobvious to go to settings and disable the hiding mode

VladoCC commented on src/main/kotlin/telegram/Handling.kt:
```diff
@@ -1179,23 +1182,40 @@ object MafiaHandler {
-                        town.updateTeams()
-                        town.prepNight()
-                        showNightRoleMenu(town, chatId, bot, -1L)
+                        if (res.isSuccess) {
```
There is a way to send simple closeable "notification" messages. Message deletion is only for specific cases. Unnecessary here.

Sevoos committed:
Time spent on altering the code: 4 minutes

VladoCC commented on src/main/kotlin/Main.kt:
```diff
@@ -42,6 +42,7 @@ const val gameDurationLimitHours = 6
 const val gameHistoryTtlHours = 24
 const val sendPendingAfterSec = 3
 const val deleteNumUpdateMsgAfterSec = 30
+const val deleteAskToDisableHidingMsgAfterSec = 15
```
Is it still used somewhere? I don't see it

Sevoos committed:
2025-07-18_03
